scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2326,6 +2329,11 @@ def _get_containers(
 
                         k8s_info["pod_info"] = pod
                         k8s_info["pod_uid"] = pod.uid
+                    else:
+                        # Pod not found by the API, including based on SCALYR_K8S_INCLUDE_ALL_CONTAINERS
+                        if not k8s_include_by_default:
```

Comment:
Do we want to log something here, e.g. Pod foo is included because `SCALYR_K8S_INCLUDE_ALL_CONTAINERS=true`.

I guess that if we do we probably want to use rate limited info / debug to avoid the overhead since I could imagine we could end up in this branch often.

Reply:
It's logged above:
` if k8s_include_by_default:
	include_message = "Including pod based on SCALYR_K8S_INCLUDE_ALL_CONTAINERS=true."
else:
	include_message = "Excluding pod based on SCALYR_K8S_INCLUDE_ALL_CONTAINERS=false."
`

I know it's far from obvious and not very readable, but I didn't want to break the code structure or refactor it.

I've added a limit key based on pod name.